examples/pytorch/eeg-gcnn/README.md

```diff
@@ -0,0 +1,48 @@
+# DGL Implementation of EEG-GCNN Paper
+This DGL example implements the EEG-GCNN model proposed in the paper [EEG-GCNN](https://arxiv.org/abs/2011.12107). The original code is [here](https://github.com/neerajwagh/eeg-gcnn).
+
+## All References
+- Paper can also be found on [PMLR](http://proceedings.mlr.press/v136/wagh20a.html).
+- [ML4H Poster](https://drive.google.com/file/d/14nuAQKiIud3p6-c8r9WLV2tAvCyRwRev/view?usp=sharing) can be helpful for understanding data preprocessing, model, and performance of the project. 
+- The recording of presentation by the author Neeraj Wagh can be found on [slideslive](https://slideslive.com/38941020/eeggcnn-augmenting-electroencephalogrambased-neurological-disease-diagnosis-using-a-domainguided-graph-convolutional-neural-network?ref=account-folder-62123-folders).
+- The slides used during the presentation can be found [here](https://drive.google.com/file/d/1dXT4QAUXKauf7CAkhrVyhR2PFUsNh4b8/view?usp=sharing).
+- Raw Data can be found in these two links: [MPI LEMON](http://fcon_1000.projects.nitrc.org/indi/retro/MPI_LEMON.html) (no registration needed), [TUH EEG Abnormal Corpus](https://www.isip.piconepress.com/projects/tuh_eeg/downloads/tuh_eeg_abnormal/) ([needs registration](https://www.isip.piconepress.com/projects/tuh_eeg/html/request_access.php))
+
+## Dependencies
+
+- Python 3.8.1
+- PyTorch 1.7.0
+- DGL 0.6.1
+- numpy 1.20.2
+- Sklearn 0.24.2
+
+## Dataset
+- Final Models, Pre-computed Features, Training Metadata can be downloaded through [FigShare](https://figshare.com/articles/software/EEG-GCNN_Supporting_Resources_for_Reproducibility/13251452).
+- In EEGGraphDataset.py, we specify the channels and electrodes and use precomputed spectral coherence values to compute the edge weights. To use this example in your own advantage, please specify your channels, electrodes and generate your own spectral coherence values. All example datasets can be downloaded from FigShare.
+## How to Run
+First, download the precomputed data, labels, indices and put them in the repo. <br>
+Then run 
+```python
+python main.py --num_feats --num_nodes --gpu_idx --num_epochs --exp_name --batch_size
+```
+
+## Performance
+|Pytorch_geometric | AUC          | Precision     | Recall       | F-1          | Bal. Accuracy |
+|------------------|--------------|---------------|--------------|--------------|---------------|
+| Shallow EEG-GCNN | 0.867(0.005) | 0.985(0.004)  | 0.680(0.023) | 0.804(0.015) | 0.802(0.006)  |
+| Deep EEG-GCNN    | 0.908(0.002) | 0.987(0.0001) | 0.753(0.007) | 0.855(0.005) | 0.842(0.004)  |
+
+|      DGL          | AUC         | Precision     | Recall       | F-1          | Bal. Accuracy |
+|-------------------|-------------|---------------|--------------|--------------|---------------|
+| Shallow EEG-GCNN  | 0.875(0.036)| 0.980(0.013)  | 0.735(0.055) | 0.839(0.035) | 0.811(0.035)  |
+| Deep EEG-GCNN     | 0.890(0.004)| 0.988(0.004)  | 0.723(0.035) | 0.834(0.022) | 0.829(0.005)  |
```

Comment:
How can a user reproduce the numbers here using `python main.py`?

Reply:
I don't think they can. We are planning to update the original repo with both pytorch_geometric and dgl implementations, and release the trainning code as well. This should be done sometime next month too. For this example, can I just point people to the original repo for reproducing the stats?

Reply:
We expect a DGL example to be self-contained and able to reproduce the results reported in README. In that case, either we can wait till you get everything ready for this PR or we can simply close the PR and add a pointer to your own repo once you get everything ready there.

Reply:
I see. Let me talk to Neeraj about this.

Reply:
Hi @mufeili, maybe the following might clarify the design of the PR:

1) We built the PR with the following assumption: that people can go to https://github.com/neerajwagh/eeg-gcnn for reproducing published results on the original data while the DGL example here would enable others to bring their own datasets (X, y) and train our model using the data they care about. Compared to our original PyT implementation, here we have skipped: a) subject-level data splitting, b) subject-level aggregate evaluation, c) cross-validation. This makes the example a "simplified" version. **Nevertheless, we can update main.py and the results table so that main.py can reproduce the new README table (as long as users pull X, y from FigShare).** We can close the PR after that change while keeping a pointer to the PyT repo for the full/"complex" implementation. We'll elaborate on this simplification in the README too so users are aware of the difference between the two.

2) We can fix the absence of norm_X in FigShare by adding normalization after X is read in the code. (instead of updating the uploaded FigShare files).

Let me know if that works for you.

Reply:
Meanwhile, the updates we make to our original PyT repo will remain independent of this example.

Reply:
Hi @neerajwagh, the proposal sounds good to me in general. To be more specific, this example does not need to fully reproduce the experiments reported in your paper and I'm good as long as a user can run the script following the instructions in the README file and get similar numbers reported there.

Reply:
@mufeili Sounds good! We'll make changes accordingly then. Thanks.